pkg/commands/apply/apply_run.go

```diff
@@ -0,0 +1,500 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package apply provides the Terraform apply functionality for Guardian.
+package apply
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/abcxyz/guardian/pkg/flags"
+	"github.com/abcxyz/guardian/pkg/github"
+	"github.com/abcxyz/guardian/pkg/storage"
+	"github.com/abcxyz/guardian/pkg/terraform"
+	"github.com/abcxyz/guardian/pkg/util"
+	"github.com/abcxyz/pkg/cli"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/sethvargo/go-githubactions"
+)
+
+const (
+	ownerReadWritePerms = 0o600
+	CommentPrefix       = "**`🔱 Guardian 🔱 APPLY`** -"
+)
+
+var _ cli.Command = (*ApplyRunCommand)(nil)
+
+// RunResult is the result of a apply operation.
+type RunResult struct {
+	commentDetails string
+}
+
+// ApplyRunCommand is a subcommand of apply and implements the cli.Command interface.
+// It performs terraform apply on the given working directory.
+type ApplyRunCommand struct {
+	cli.BaseCommand
+
+	cfg *Config
+
+	directory                 string
+	childPath                 string
+	planFilename              string
+	gitHubLogURL              string
+	computedPullRequestNumber int
+
+	flags.GitHubFlags
+	flags.RetryFlags
+
+	flagBucketName           string
+	flagCommitSHA            string
+	flagPullRequestNumber    int
+	flagAllowLockfileChanges bool
+	flagLockTimeout          time.Duration
+
+	actions         *githubactions.Action
+	githubClient    github.GitHub
+	storageClient   storage.Storage
+	terraformClient terraform.Terraform
+}
+
+// Desc provides a short, one-line description of the command.
+func (c *ApplyRunCommand) Desc() string {
+	return "Run the Terraform apply for a directory"
+}
+
+// Help is the long-form help output to include usage instructions and flag
+// information.
+func (c *ApplyRunCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options] <directory>
+
+	Run the Terraform apply for a directory.
+`
+}
+
+func (c *ApplyRunCommand) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+
+	c.GitHubFlags.Register(set)
+	c.RetryFlags.Register(set)
+
+	f := set.NewSection("COMMAND OPTIONS")
+
+	f.IntVar(&cli.IntVar{
+		Name:    "pull-request-number",
+		Target:  &c.flagPullRequestNumber,
+		Example: "100",
+		Usage:   "The GitHub pull request number associated with this apply run.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "commit-sha",
+		Target:  &c.flagCommitSHA,
+		Example: "100",
```

Comment:
"100" seems like an unrealistic example of a SHA (maybe this was a copy-paste oversight?). How about some funky hex string instead?